plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,538 @@
+            ["static_report_route_map_uuid", "static_report_route_map"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    version3_asm = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("version3_asm"))
+    fast_leave = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("fast_leave"))
+    report_link_local_groups = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("report_link_local_groups"))
+    igmp_version = module.params.get("igmp_version")
+    group_timeout = module.params.get("group_timeout")
+    query_interval = module.params.get("query_interval")
+    query_response_interval = module.params.get("query_response_interval")
+    last_member_count = module.params.get("last_member_count")
+    last_member_response_time = module.params.get("last_member_response_time")
+    startup_query_count = module.params.get("startup_query_count")
+    startup_query_interval = module.params.get("startup_query_interval")
+    querier_timeout = module.params.get("querier_timeout")
+    robustness_variable = module.params.get("robustness_variable")
+    state_limit_route_map_uuid = module.params.get("state_limit_route_map_uuid")
+    state_limit_route_map = module.params.get("state_limit_route_map")
+    report_policy_route_map_uuid = module.params.get("report_policy_route_map_uuid")
+    report_policy_route_map = module.params.get("report_policy_route_map")
+    static_report_route_map_uuid = module.params.get("static_report_route_map_uuid")
+    static_report_route_map = module.params.get("static_report_route_map")
+    maximum_multicast_entries = module.params.get("maximum_multicast_entries")
+    reserved_multicast_entries = module.params.get("reserved_multicast_entries")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "tenant", template, template_id)
+    mso_template.validate_template("tenantPolicy")
+    ops = []
+    empty_state_limit_route_map = False
+    empty_report_policy_route_map = False
+    empty_static_report_route_map = False
+
+    existing_igmp_interface_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("igmpInterfacePolicies", [])
+    object_description = "IGMP Interface Policy"
+    path = "/tenantPolicyTemplate/template/igmpInterfacePolicies"
+
+    if uuid or name:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_igmp_interface_policies,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            igmp_interface_policy_attrs_path = "{0}/{1}".format(path, match.index)
+            set_names_for_references(mso, match.details)
```

Comment:
Once my PR is reviewed by everyone, I would make the following change: https://github.com/CiscoDevNet/ansible-mso/pull/615/files#:~:text=def%20set_names_for_references(mso%2C%20payload%2C%20reference_dict)%3A

Reply:
changed